From aa988afb009307c6b854b5da366d34e0ee2ea763 Mon Sep 17 00:00:00 2001
From: Marek Kasik <mkasik@redhat.com>
Date: Mon, 19 May 2014 14:32:19 +0200
Subject: Scroll to the search result selected by user

Scroll to page on which is the search result selected by user
when not in continuous mode.

https://bugzilla.gnome.org/show_bug.cgi?id=730252

diff --git a/libview/ev-view.c b/libview/ev-view.c
index 5ed2d78..f3aa033 100644
--- a/libview/ev-view.c
+++ b/libview/ev-view.c
@@ -8158,6 +8158,9 @@ jump_to_find_page (EvView *view, EvViewFindDirection direction, gint shift)
 			break;
 		}
 	}
+
+	if (!view->continuous)
+		ev_document_model_set_page (view->model, view->find_page);
 }
 
 static void
@@ -8282,6 +8285,7 @@ ev_view_find_set_result (EvView *view, gint page, gint result)
 {
 	view->find_page = page;
 	view->find_result = result;
+	jump_to_find_page (view, EV_VIEW_FIND_NEXT, 0);
 	jump_to_find_result (view);
 	gtk_widget_queue_draw (GTK_WIDGET (view));
 }
-- 
cgit v0.10.1

